src/IO/S3/copyS3File.cpp

```diff
@@ -91,20 +91,24 @@ namespace
 
         struct UploadPartTask
         {
-            std::unique_ptr<Aws::AmazonWebServiceRequest> req;
```

Comment:
Here was a problem. The `req` contains a `ReadBufferFromS3`, and so keeping `req` in all the tasks means keeping all such read buffers until the end of copying all the parts (and there can be thousands of parts). Each `ReadBufferFromS3` normally takes 1MB of memory so keeping read buffers from already uploaded parts means consuming multiple gigabytes without much sense.

This PR is fixing that - now `req` is created and destroyed while running each task in `processUploadTask()`.

Reply:
nice description, probably add it into the src file as comments

Reply:
done